assets/js/phoenix_live_view/live_socket.js

```diff
@@ -523,12 +525,11 @@ export default class LiveSocket {
     this.bindClicks()
     if(!dead){ this.bindForms() }
     this.bind({keyup: "keyup", keydown: "keydown"}, (e, type, view, targetEl, phxEvent, eventTarget) => {
-      let matchKey = targetEl.getAttribute(this.binding(PHX_KEY))
-      let pressedKey = e.key && e.key.toLowerCase() // chrome clicked autocompletes send a keydown without key
-      if(matchKey && matchKey.toLowerCase() !== pressedKey){ return }
-
-      let data = {key: e.key, ...this.eventMeta(type, e, targetEl)}
-      JS.exec(type, phxEvent, view, targetEl, ["push", {data}])
+      let matchKey = parsePhxKey(targetEl.getAttribute(this.binding(PHX_KEY)))
```

Comment:
Maybe @chrismccord changed his mind but IIRC he has not a big fan of relying on regexes for matching attributes, given the cost it implies on every key press or click.

Reply:
Thats a good point, I will ask

Reply:
I believe alpine doesn't pay this cost because they don't have to deal with dynamic phx-key bindings, they compile once on load and are done with it. I could switch to .split(".") which would give near identical result.

Reply:
I could also memoize it